components/Custom/FormComponents/Form/index.tsx

```diff
@@ -16,7 +16,7 @@ const Form = <T,>({
 }: PropsWithChildren<HookFormProps<T>>) => {
   const methods = useForm({
     defaultValues,
-    mode: 'onSubmit'
+    mode: 'onChange'
```

Comment:
I feel instead of having it as `onChange`, change it back to `onSubmit` and do the check when the user submits the form. This should save network calls and is better for performance